api/rest/cluster_bundle_handler_test.go

```diff
@@ -871,6 +871,24 @@ func TestRemoteBundleCreation(t *testing.T) {
 	})
 }
 
+func TestWorkDirIsCreatedIfNotExists(t *testing.T) {
+	t.Parallel()
+
+	workdir, err := ioutil.TempDir("", "work-dir")
+	require.NoError(t, err)
+	err = os.RemoveAll(workdir)
+	require.NoError(t, err)
+
+	coord := mockCoordinator{}
+	client := &MockClient{}
+	tools := &MockedTools{}
+	urlBuilder := MockURLBuilder{}
+	_, err = NewClusterBundleHandler(coord, client, tools, workdir, time.Millisecond, urlBuilder)
+	require.NoError(t, err)
+
+	assert.DirExists(t, workdir)
+}
+
```

Comment:
I one more test to check what happens when file under given workdir path exists
```suggestion

func TestBundleDirCreationFailsWhenFileWithTheSameNameExists(t *testing.T) {
	t.Parallel()

	workdir, err := ioutil.TempFile("", "work-dir")
	require.NoError(t, err)
	_, err = NewBundleHandler(workdir.Name(), nil, time.Millisecond)
	assert.Error(t, err)
}
```